nrf_rpc/include/nrf_rpc_tr.h

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#ifndef NRF_RPC_TR_H_
+#define NRF_RPC_TR_H_
+
+#if defined(CONFIG_NRF_RPC_TR_RPMSG)
```

Comment:
But `nrf_rpc_rpmsg` is not the only transport that we are planning to support. That's why we added Kconfig options to easily switch the transport.

`nrf_rpc_os.h` is a different story. It is not possible to simply switch the OS. nRF RPC must be ported to specific OS e.g. how we did it for Zephyr in https://github.com/nrfconnect/sdk-nrf/pull/2179

That's why I want to keep current approach.

Reply:
Shouldn't the entire port be covered by the header `nrf_rpc_os.h` though? So atleast to my understand it would be the same as changing the transport layer.

But if you wish to keep it like this, then keep it.